src/main/java/com/google/cloud/spanner/pgadapter/parsers/ArrayParser.java

```diff
@@ -28,29 +32,66 @@
  * Translate wire protocol to array. Since arrays house any other specified types (including
  * potentially arrays), we use all parser types to parse each item within.
  */
-public class ArrayParser extends Parser<List> {
+public class ArrayParser extends Parser<List<?>> {
+  private static final String STRING_TOGGLE = "\"";
+  private static final String ARRAY_DELIMITER = ",";
+  private static final String PG_ARRAY_OPEN = "{", PG_ARRAY_CLOSE = "}";
+  private static final String SPANNER_ARRAY_OPEN = "[", SPANNER_ARRAY_CLOSE = "]";
 
-  private static String STRING_TOGGLE = "\"";
-  private static String ARRAY_DELIMITER = ",";
-  private static String PG_ARRAY_OPEN = "{", PG_ARRAY_CLOSE = "}";
-  private static String SPANNER_ARRAY_OPEN = "[", SPANNER_ARRAY_CLOSE = "]";
+  private final Type arrayElementType;
+  private final boolean isStringEquivalent;
 
-  int arrayType;
-  boolean isStringEquivalent;
-
-  public ArrayParser(ResultSet item, int position) throws SQLException {
+  public ArrayParser(ResultSet item, int position) {
```

Comment:
As we are now only throwing `SpannerException` instead of `SQLException` we do not need to declare the exception, as `SpannerException` is a `RuntimeException` instead of a checked `Exception`. Although it is possible to declare a method as throwing a `RuntimeException`, it is not common to do so. For example all methods in the Java client library could throw a `SpannerException`, but these are also not declared.